corehq/apps/app_manager/views/download.py

```diff
@@ -255,14 +255,17 @@ def resolve_path(path):
         return RegexURLResolver(
             r'^', 'corehq.apps.app_manager.download_urls').resolve(path)
 
+    def create_build_files(build_profile_id=None):
+        request.app.create_build_files(build_profile_id=build_profile_id)
+        request.app.save()
+
     def create_build_files_if_necessary_handling_conflicts(is_retry=False):
         try:
             try:
                 # look for file guaranteed to exist if profile is created
                 request.app.fetch_attachment('files/{id}/profile.xml'.format(id=build_profile))
             except ResourceNotFound:
-                request.app.create_build_files(build_profile_id=build_profile)
-                request.app.save()
+                create_build_files(build_profile)
```

Comment:
You didn't introduce this, but it'd be nice to rename `build_profile` to `build_profile_id`.

Reply:
https://github.com/dimagi/commcare-hq/pull/24751/commits/08dba3d7ec8c051ee0584e7b7c326a42ac9dd64d

Reply:
way to leave the code cleaner than you found it 🙂